jawampa-core/src/main/java/ws/wamp/jawampa/WampClient.java

```diff
@@ -333,7 +335,7 @@ public void run() {
      * of the call be Observable&lt;String&gt;.
      * @return An observable that can be used to subscribe on the topic.
      */
-    public <T> Observable<T> makeSubscription(final String topic, final Class<T> eventClass) {
+    public <T> Observable<MessageInfo<T>> makeSubscription(final String topic, final Class<T> eventClass) {
```

Comment:
White I agree that getting the topic is helpful for some users (like you), it makes things much more complicated for other users. And we would especially break the existing API.
My proposal would be to add 2 extra overloads which allow to retrieve the MesageInfo/EventDetails. They would need a new name because they would only differentiate in the return value. Maybe add 2 makeSubscriptionWithDetails functions that behave as your version of the methods and leave the existing ones as they are.